staging/src/k8s.io/kubelet/pkg/cri/streaming/remotecommand/httpstream_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remotecommand
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+func TestHandleResizeEvents(t *testing.T) {
+	var testTerminalSize remotecommand.TerminalSize
+	rawTerminalSize, err := json.Marshal(&testTerminalSize)
+	require.NoError(t, err)
+
+	testCases := []struct {
+		name             string
+		resizeStreamData []byte
+		cancelContext    bool
+		readFromChannel  bool
+	}{
+		{
+			name:             "data attempted to be sent on the channel; channel not read; context canceled",
+			resizeStreamData: rawTerminalSize,
+			cancelContext:    true,
+		},
+		{
+			name:             "data attempted to be sent on the channel; channel read; context not canceled",
+			resizeStreamData: rawTerminalSize,
+			readFromChannel:  true,
+		},
+		{
+			name:          "no data attempted to be sent on the channel; context canceled",
+			cancelContext: true,
+		},
+		{
+			name: "no data attempted to be sent on the channel; context not canceled",
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
Please, consider using [ktesting.Init(t)](https://github.com/kubernetes/kubernetes/blob/master/test/utils/ktesting/tcontext.go#L243)t instead of `context.TODO`, thanks!

Reply:
could you please explain why you think it would be helpful in this case?

Reply:
thanks, but I disagree. All we need here is an empty "dummy" context to build our cancellation context with. `context.TODO()` is such context and is used heavily in other tests and across kubernetes testing suite in general. An alternative could be `context.Background()`, which is explicitly mentioned by the same golang documentation for potential usage in tests.

`ktesting.Init()` however comes with additional features which IMO we just don't need here. It also sets up some goroutines (via `init()` when package is imported and by `Init()` explicitly) which makes validating goroutine leaks that much difficult and complex.

Reply:
changed